src/main/java/io/excaliburfrc/robot/subsystems/Drive.java

```diff
@@ -19,15 +43,112 @@
   private final CANSparkMax rightFollower =
       new CANSparkMax(DrivetrainConstants.RIGHT_FOLLOWER_ID, MotorType.kBrushless);
 
+  private final RelativeEncoder rightEncoder = rightLeader.getEncoder();
+  private final RelativeEncoder leftEncoder = leftLeader.getEncoder();
+
   private final DifferentialDrive drive = new DifferentialDrive(leftLeader, rightLeader);
 
+  private final DifferentialDriveKinematics driveKinematics =
+      new DifferentialDriveKinematics(DrivetrainConstants.TRACKWIDTH_METERS);
+
+  private final DifferentialDriveVoltageConstraint autoVoltageConstraint =
+      new DifferentialDriveVoltageConstraint(
+          new SimpleMotorFeedforward(kS, kV, kA), driveKinematics, 10);
+  private final TrajectoryConfig config =
+      new TrajectoryConfig(
+              DrivetrainConstants.MAX_SPEED_METERS_PER_SECOND,
+              DrivetrainConstants.MAX_ACCELERATION_METERS_PER_SECOND_SQUARED)
+          .setKinematics(driveKinematics)
+          .addConstraint(autoVoltageConstraint);
+  public final Trajectory trajectory =
+      TrajectoryGenerator.generateTrajectory(
+          new Pose2d(0, 0, new Rotation2d(0)),
+          List.of(new Translation2d(1, 1), new Translation2d(2, -1)),
+          new Pose2d(3, 0, new Rotation2d(0)),
+          config);
```

Comment:
All three of these (constraint, config, and trajectory) shouldn't be in the Drive subsystem.

The concept is to have a method that returns a command (group) that follows a supplied trajectory.